### PR TITLE
Fixes #19496 - passenger recycler

### PR DESCRIPTION
**Pull request by swapab**
@lzap, I am carrying over https://github.com/theforeman/foreman_maintain/pull/78 here

Following things covered in this PR:
1. Updated `bin/passenger-recycler` to comply with Rubocop standards.
2. Introduced `extras` directory. Which can be used to reside files such as `.cron, .conf` etc.
3. New procedure to execute `passenger-recycler`
   `./bin/foreman-maintain advanced procedure run passenger-recycler`

Next steps:
1. Update [foreman-packaging](https://github.com/theforeman/foreman-packaging/tree/rpm/develop) to install `bin/passenger-recycler` to `/usr/bin/`
2. Setup ` extras/passenger-recycler` to ` /etc/cron.d/passenger-recycler`.

PS: I have not renamed the `passenger-recycler` script and thinking to keep it that way.

### CODE REVIEW

**Comment by lzap** [comment]
LGTM, note you have `passenger-recycler.cron` twice, perhaps a leftover?

**Comment by swapab** [comment]
Deleted `config/passenger-recycler.cron`. Thanks!

**Comment by lzap** [comment]
LGTM one nitpick - I realized that 5 minutes by default is too tight, we max with 4 SIGTERMs with 90 seconds delay each which is in total 6 minutes. In the worst case, another process would be spawned. I suggest to run the cron job every 15 minutes by default, can you change the cronjob and also `passenger-recycler.yaml` comments to 15 mins?

**Comment by swapab** [comment]
Done. @lzap Can we merge this branch into master? As I will be doing the relevant packaging changes and bump the version.

**Comment by lzap** [comment]
Sorry for late reponse, absolutely, I havent tested this version myself tho.

**Comment by swapab** [comment]
@iNecas Can you please review this PR once and merge accordingly?

File: extras/passenger-recycler.cron [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Configuration file /etc/cron.d/passenger-recycler to run passenger-recycler
+# every 5 minutes.
```
Comment (lzap) [inline extras/passenger-recycler.cron]: Nitpick - it is every 15 minutes now.

Comment (swapab) [reply extras/passenger-recycler.cron]: ACK

**Comment by swapab** [comment]
Thanks, @iNecas I'll get the packaging changes ready.

File: bin/passenger-recycler [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+exit 0 unless CONFIG[:ENABLED]
+
+def running?(pid)
+  return Process.getpgid(pid) != -1
+rescue Errno::ESRCH
+  return false
+end
+
+def debug(msg)
+  puts(msg) if CONFIG[:DEBUG]
```
Comment (ohadlevy) [inline bin/passenger-recycler]: nitpick: probably better to output to STDOUT (using warn vs puts)

**Pull request merged**